spec/rspec/matchers/dsl_spec.rb

```diff
@@ -563,6 +563,28 @@ def foo
         expect(self.matcher.matches?('other value')).to be_falsey
       end
 
+      context "with a negated match block" do
+        let(:matcher) do
+          new_matcher(:name, "value") do |expected|
+            match do |actual|
+              expect(actual).to eq expected
+            end
+
+            match_when_negated do |actual|
+              expect(actual).not_to eq expected
+            end
+          end
+        end
+
+        it "returns true if the wrapped expectation passes" do
+          expect(self.matcher.does_not_match?('other value')).to be_truthy
+        end
```

Comment:
Finally getting around to finishing this PR. Thanks for your patience!

@myronmarston, your speculation here is exactly what I said in [this comment](https://github.com/rspec/rspec-expectations/pull/789#discussion_r30530471). ;) I'm glad you came to the same conclusion I did separately as it's a very head scratching thing. I'll see what I can do about adding specs to make sure we're good but this _does_ seem like if its wrong things should be _quite_ broken in master(?). The behavior I outlined at the end of the comment I mention above is still true in latest master (`expect("foo").not_to eq "bar" => false`)